src/main/kotlin/com/team4099/robot2023/subsystems/groundintake/GroundIntake.kt

```diff
@@ -260,6 +252,12 @@ class GroundIntake(private val io: GroundIntakeIO) : SubsystemBase() {
     // current state to the next state ensures that we run the logic for the state we want in the
     // next loop cycle.
     currentState = nextState
+
+    if (requestedState is GroundIntakeRequest.OpenLoop) {
```

Comment:
the thing I was saying earlier was if you make `requestedState` (or whatever you rename it to) a local variable here (`val myCurrentRequest = this.currentRequest` or something with better naming), Kotln will smart cast here so you don't need the `as` stuff